lib/tarantool_server.py

```diff
@@ -38,12 +40,15 @@
 def save_join(green_obj, timeout=None):
     """
     Gevent join wrapper for
-    test-run stop-on-crash feature
+    test-run stop-on-crash/stop-on-timeout feature
 
-    :return True in case of crash and False otherwise
+    :return True in case of crash or test timeout and False otherwise
     """
     try:
-        green_obj.join(timeout=timeout)
+        green_obj.get(timeout=timeout)
+    except Timeout:
+        color_stdout("Test timeout of %d secs reached\t" % timeout, schema='error')
```

Comment:
We should kill the greenlet in the case, otherwise it may write to a temporary result file of a new run of the same test. I saw it on a regular run of all tests: replication/election_qsync_stress.test.lua was run twice (memtx and vinyl) on the same worker with different configurations (I guess it has big probability due to sequencial run of fragile tests).

Reply:
Implemented.